File: livebox-common/src/main/java/com/creations/livebox_common/adapter/ObservableAdapter.java
package com.creations.livebox_common.adapter;

import io.reactivex.Observable;

/**
 * @author Sérgio Serra on 27/08/2018.
 * Criations
 * sergioserra99@gmail.com
 */
public interface ObservableAdapter<T, R> {
    R adapt(Observable<T> observable);
}

File: livebox-common/src/main/java/com/creations/livebox_common/adapter/ObservableAdapter.kt
package com.creations.livebox_common.adapter

import io.reactivex.Observable

/**
 * @author Sérgio Serra on 27/08/2018.
 * sergioserra99@gmail.com
 */
interface ObservableAdapter<T, R> {
    fun adapt(observable: Observable<T>): R
}
